### DISCUSSION TITLE
Error: Permission denied for table XXXX

### DISCUSSION THREAD

**Discussion by aws-avocado** [supabase/supabase#10671] [Questions]
I'm currently trying to access my own row in a table with this JS code in NextJS 12 with the SDK:

`const getUserDetails = () => supabase.from(‘users’).select(‘*’).single()`

However I get this error of a "permission denied for table XXX" even though I've already set the RLS for that particular table, with "USING (uid() = uid)"

This same code and RLS works fine on my staging project. Is there something I'm missing here, or a table reload that I need to do?

**Comment by silentworks** [comment]
Where are you running this function? is it in the client side code of NextJS or is it in the server side code?
